matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncRequestStateTracker.kt

```diff
@@ -16,23 +16,26 @@
 
 package org.matrix.android.sdk.internal.session.sync
 
-import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
 import org.matrix.android.sdk.api.session.sync.InitialSyncStep
 import org.matrix.android.sdk.api.session.sync.SyncRequestState
 import org.matrix.android.sdk.internal.session.SessionScope
 import javax.inject.Inject
 
 @SessionScope
-internal class SyncRequestStateTracker @Inject constructor() :
-        ProgressReporter {
+internal class SyncRequestStateTracker @Inject constructor(
+        private val coroutineScope: CoroutineScope
```

Comment:
Note that the CoroutineScope [from the crypto module](https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/CryptoModule.kt#L135) will be injected here.

This is not a big deal, I think, but this will have to be re-worked (by us)

Reply:
(and this is tracked by #6432)